client/src/components/EmbeddingPanel/util.ts

```diff
@@ -0,0 +1,6 @@
+export const margin = { top: 40, right: 5, bottom: 20, left: 60 };
+export const width = 400;
+export const height = 400 - margin.top - margin.bottom;
+export const innerHeight = height - 2;
+
+export const devicePixelRatio = window.devicePixelRatio || 1;
```

Comment:
I've found this line defined several times throughout, maybe worth moving it to globals?